modules/backend/formwidgets/IconPicker.php

```diff
@@ -0,0 +1,1635 @@
+<?php namespace Backend\FormWidgets;
+
+use Url;
+use Config;
+use Backend\Classes\FormWidgetBase;
+use Winter\Storm\Exception\ApplicationException;
+
+/**
+ * Icon picker
+ * Renders an icon picker field.
+ *
+ * @package winter\wn-backend-module
+ * @author Robert Alexa, Jack Wilkinson
+ */
+class IconPicker extends FormWidgetBase
+{
+    /**
+     * @inheritDoc
+     */
+    protected $defaultAlias = 'iconpicker';
+
+    protected array $fontLibraries = [
```

Comment:
These should be defined in a separate YAML file / files and be configurable by the user

Reply:
How I envisioned this is the following:
- WinterCMS provides a default list of icons based on all available FontAwesome icons at the current version.
- The end user is able to manipulate this default as desired by hooking into the `backend.formwidgets.iconpicker.fontLibraries` event.
- The default list of icons should be moved out of where it currently is. Instead, Font Awesome should be updated to the last known version, all shims be created, and this "library" by generated there and then based on these available icons. This generated library should be included in the iconpicker instead of a hardcoded list. This way, at the next FA update you simply regenerate the "library" and job is good.

In the meantime I have added a comment above the current location.

Reply:
Further changes:
- moved it into a new config file
- retired the event as people can just overload the config by doing their own one in their config folder